source/usr/lib/mcvirt/virtual_machine/virtual_machine.py

```diff
@@ -946,255 +919,147 @@ def duplicate(self, mcvirt_instance, duplicate_vm_name):
-                                                 ' is not initialised')
-
-        # Determine if VM already exists
-        if (VirtualMachine._checkExists(mcvirt_instance, name)):
-            raise VmAlreadyExistsException('Error: VM already exists')
-
-        # If a node has not been specified, assume the local node
-        if (node is None):
-            node = Cluster.getHostname()
-
-        # If DRBD has been chosen as a storage type, ensure it is enabled on the node
-        from mcvirt.node.drbd import DRBD as NodeDRBD, DRBDNotEnabledOnNode
-        if (storage_type == 'DRBD' and not NodeDRBD.isEnabled()):
-            raise DRBDNotEnabledOnNode('DRBD is not enabled on this node')
-
-        # Create directory for VM on the local and remote nodes
-        if (os.path.exists(VirtualMachine.getVMDir(name))):
-            raise VmDirectoryAlreadyExistsException('Error: VM directory already exists')
-
-        # If available nodes has not been passed, assume the local machine is the only
-        # available node if local storage is being used. Use the machines in the cluster
-        # if DRBD is being used
-        cluster_object = Cluster(mcvirt_instance)
-        all_nodes = cluster_object.getNodes()
-        all_nodes.append(Cluster.getHostname())
-        if (len(available_nodes) == 0):
-            if (storage_type == 'DRBD' and mcvirt_instance.initialiseNodes()):
-                # If the available nodes are not specified, use the
-                # nodes in the cluster
-                available_nodes = all_nodes
-            else:
-                # For local VMs, only use the local node as the available nodes
-                available_nodes = [Cluster.getHostname()]
-
-        # If there are more than the maximum number of DRBD machines in the cluster,
-        # add an option that forces the user to specify the nodes for the DRBD VM
-        # to be added to
-        if (storage_type == 'DRBD' and len(available_nodes) != NodeDRBD.CLUSTER_SIZE):
-            raise MCVirtException('Exactly two nodes must be specified')
-
-        for check_node in available_nodes:
-            if (check_node not in all_nodes):
-                raise NodeDoesNotExistException('Node \'%s\' does not exist' % check_node)
-
-        if (Cluster.getHostname() not in available_nodes and mcvirt_instance.initialiseNodes()):
-            raise MCVirtException('One of the nodes must be the local node')
-
-        # Create directory for VM
-        os.makedirs(VirtualMachine.getVMDir(name))
+        if self.getStorageType() == 'Local':
+            remote_node = cluster_instance.get_remote_node(destination_node)
+            remote_vm_factory = remote_node.get_connection('virtual_machine_factory')
+            remote_vm = remote_vm_factory.getVirtualMachineByName(self.get_name())
+            remote_node.annotate_object(remote_vm)
+            remote_vm.register()
```

Comment:
There are two if statements with the same condition right next to each other here (``if self.getStorageType() == 'Local':``) - couldn't this just be one if?